gumroad/src/gumroad-block/block.js

```diff
@@ -24,7 +24,7 @@ const {
 
 const gumroadURL = 'https://gumroad.com/l/demo';
 const defaultID = 'demo';
-const gumroadURLRegex = /(https:\/\/gumroad\.com\/l\/)|(https:\/\/gumroad\.com\/)|(https:\/\/gum\.co\/l\/)|(https:\/\/gum\.co\/)/g;
+const gumroadURLRegex = /(https:\/\/gumroad\.com\/a\/.+\/)|(https:\/\/gumroad\.com\/l\/)|(https:\/\/gumroad\.com\/)|(https:\/\/gum\.co\/l\/)|(https:\/\/gum\.co\/)/g;
```

Comment:
Regex change and dropped `(https:\/\/gum\.co\/l\/)` -

```suggestion
const gumroadURLRegex = /(https:\/\/gumroad\.com\/a\/[[:alnum:]]+\/)|(https:\/\/gumroad\.com\/l\/)|(https:\/\/gumroad\.com\/)|(https:\/\/gum\.co\/)/g;
```